### PR TITLE
add urlOrNull method in router and test

### PR DESCRIPTION
**Pull request by EliasDeVos**
pull request for the following issue #11625,

I didn't know if i had to copy the docblocks for usage from `Router::url` or just refer to them.
I copied them now but this can be changed

### CODE REVIEW

File: src/Routing/Router.php [hunk 1]
```diff
@@ -682,6 +682,54 @@ public static function url($url = null, $full = false)
...
+     *   string.
+     * @param bool $full If true, the full base URL will be prepended to the result.
+     *   Default is false.
+     * @return string Full translated URL with base path or null if url does not exist
+     */
+    public static function urlOrNull($url = null, $full = false)
+    {
+        try {
+            return static::url($url, $full);
+        } catch (\Cake\Routing\Exception\MissingRouteException $e) {
```
Comment (ADmad) [inline src/Routing/Router.php]: Add a "use" at top of file instead of using FQCN.

**Review by ADmad — CHANGES_REQUESTED** [review]

File: tests/TestCase/Routing/RouterTest.php [hunk 2]
```diff
@@ -129,6 +129,19 @@ public function testRouteDefaultParams()
...
+     * testRouteUrlOrNull method
+     *
+     * @return void
+     */
+    public function testRouteUrlOrNull()
+    {
+        Router::connect('/:controller/:action', ['controller' => 'posts']);
+        $this->assertEquals(Router::urlOrNull(['action' => 'view']), '/view');
+
+        $this->assertEquals(Router::urlOrNull(['action' => 'view', 'controller' => 'users', 'plugin' => 'test']), null);
```
Comment (ADmad) [inline tests/TestCase/Routing/RouterTest.php]: Use `assertNull()`

**Review by ADmad — CHANGES_REQUESTED** [review]

File: src/Routing/Router.php [hunk 3]
```diff
@@ -682,6 +683,54 @@ public static function url($url = null, $full = false)
...
+     *
+     * @param string|array|null $url An array specifying any of the following:
+     *   'controller', 'action', 'plugin' additionally, you can provide routed
+     *   elements or query string parameters. If string it can be name any valid url
+     *   string.
+     * @param bool $full If true, the full base URL will be prepended to the result.
+     *   Default is false.
+     * @return string Full translated URL with base path or null if url does not exist
+     */
+    public static function urlOrNull($url = null, $full = false)
```
Comment (dereuromark) [inline src/Routing/Router.php]: Why would the first argument be nullable?

Comment (EliasDeVos) [reply src/Routing/Router.php]: I just took the same default values that are passed to `Router:;url()` but i could remove the nullable for $url

Comment (EliasDeVos) [reply src/Routing/Router.php]: @dereuromark yes that was the idea :P

File: src/Routing/Router.php [hunk 4]
```diff
@@ -682,6 +683,54 @@ public static function url($url = null, $full = false)
...
+     *   cake relative URLs are required when using requestAction.
+     * - `_scheme` - Set to create links on different schemes like `webcal` or `ftp`. Defaults
+     *   to the current scheme.
+     * - `_host` - Set the host to use for the link. Defaults to the current host.
+     * - `_port` - Set the port if you need to create links on non-standard ports.
+     * - `_full` - If true output of `Router::fullBaseUrl()` will be prepended to generated URLs.
+     * - `#` - Allows you to set URL hash fragments.
+     * - `_ssl` - Set to true to convert the generated URL to https, or false to force http.
+     * - `_name` - Name of route. If you have setup named routes you can use this key
+     *   to specify it.
```
Comment (markstory) [inline src/Routing/Router.php]: Instead of duplicating the options you should use `@see Router::url()`

**Comment by markstory** [comment]
Is getting a string or null back the API we want to provide? I originally thought we were going for something like

```php
if (Router::routeExists($url)) {
  // Do something, possibly call Router::url($url);
}
```

The difference is returning a `bool` vs `string|null`

**Comment by EliasDeVos** [comment]
@markstory i changed it, now it checks and returns true or false

**Comment by markstory** [comment]
Thank you @EliasDeVos :clap:

**Pull request merged**